src/m365/spo/commands/site/site-admin-list.ts

```diff
@@ -0,0 +1,262 @@
+import request, { CliRequestOptions } from '../../../../request.js';
+import { formatting } from '../../../../utils/formatting.js';
+import { ClientSvcResponse, ClientSvcResponseContents, FormDigestInfo, spo } from '../../../../utils/spo.js';
+import { validation } from '../../../../utils/validation.js';
+import SpoCommand from '../../../base/SpoCommand.js';
+import commands from '../../commands.js';
+import { ListPrincipalType } from '../list/ListPrincipalType.js';
+import { SPOSitePropertiesEnumerable } from './SPOSitePropertiesEnumerable.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface AdminUserResult {
+  email: string;
+  loginName: string;
+  name: string;
+  userPrincipalName: string;
+}
+
+interface AdminResult {
+  value: AdminUserResult[];
+}
+
+interface SiteUserResult {
+  Email: string;
+  Id: number;
+  IsSiteAdmin: boolean;
+  LoginName: string;
+  PrincipalType: number;
+  Title: string;
+}
+
+interface SiteResult {
+  value: SiteUserResult[];
+}
+
+interface CommandResultItem {
+  Id?: number;
+  Email: string;
+  IsPrimaryAdmin: boolean;
+  LoginName: string;
+  Title: string;
+  PrincipalType?: number;
+  PrincipalTypeString?: string;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  asAdmin?: boolean;
+}
+
+class SpoSiteAdminListCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_ADMIN_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all administrators of a specific SharePoint site';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        siteUrl: typeof args.options.siteUrl !== 'undefined',
+        asAdmin: typeof args.options.asAdmin !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --siteUrl <siteUrl>'
+      },
+      {
+        option: '--asAdmin'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => validation.isValidSharePointUrl(args.options.siteUrl)
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      if (args.options.asAdmin) {
+        await this.callActionAsAdmin(logger, args);
+        return;
+      }
+
+      await this.callAction(logger, args);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async callActionAsAdmin(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      await logger.logToStderr('Retrieving site administrators as an administrator...');
+      const adminUrl: string = await spo.getSpoAdminUrl(logger, this.debug);
+      const siteId = await this.getSiteIdBasedOnUrl(logger, args.options.siteUrl, adminUrl);
```

Comment:
@mkm17 just to double check. if you are a tenant admin isn't it possible to get the site ID using MSGraph and here we could use the `spo.getSiteId()` util for that? or am I missing something here 🤔

Reply:
You were right, this one can be also used. :) I found the previous method somewhere in the CLI code, and looked also ok.